### PR TITLE
Feature/#445

### PR DESCRIPTION
**Pull request by s0cks**
### Description of the Change

This should allow sorting of instances by with a variety of different sorting strategies

### Testing

Lightly tested locally. No unit tests, will add if wanted.

### Related Issues

Fixes: #445

### CODE REVIEW

**Review by RyanTheAllmighty — CHANGES_REQUESTED** [review]
Thanks for this. And welcome back, hope all has been well.

Looks like builds failing due to missing license headers, just run `./gradlew updateLicenses` to update them and commit them back up.

File: src/main/java/com/atlauncher/Gsons.java [hunk 1]
```diff
@@ -18,14 +18,10 @@
...
+import java.time.Instant;
 import java.util.Date;
 
-import com.atlauncher.data.AbstractAccount;
-import com.atlauncher.data.AccountTypeAdapter;
-import com.atlauncher.data.ColorTypeAdapter;
-import com.atlauncher.data.DateTypeAdapter;
-import com.atlauncher.data.PackVersion;
-import com.atlauncher.data.PackVersionTypeAdapter;
+import com.atlauncher.data.*;
```
[the remark this answers is not in the document]
Comment (s0cks) [reply src/main/java/com/atlauncher/Gsons.java:24]: Sure thing, I just installed Idea on this laptop so still sorting out the little things
~ Tazz

File: src/main/java/com/atlauncher/gui/components/CollapsiblePanel.java [hunk 2]
```diff
@@ -127,6 +131,11 @@ public CollapsiblePanel(Instance instance) {
             title = instance.launcher.name;
         }
 
+        final DateTimeFormatter formatter = DateTimeFormatter.ofLocalizedDateTime(FormatStyle.SHORT)
```
[the remark this answers is not in the document]
Comment (s0cks) [reply src/main/java/com/atlauncher/gui/components/CollapsiblePanel.java:134]: Honestly this wasnt supposed to make it in the PR, but if you'd like to change the title sure thing. 

I was using it to test the sorting.
~ Tazz

File: src/main/java/com/atlauncher/gui/tabs/InstancesTab.java [hunk 3]
```diff
@@ -187,4 +211,14 @@ public void onRelocalization() {
             nilCard.setMessage(GetText.tr("There are no instances to display.\n\nInstall one from the Packs tab."));
         }
     }
+
+    @Override
+    public void itemStateChanged(ItemEvent e){
```
[the remark this answers is not in the document]
Comment (s0cks) [reply src/main/java/com/atlauncher/gui/tabs/InstancesTab.java:216]: - Less generated/anonymous classes
- Somewhat cleaner 
while still providing access to the instance variables
~ Tazz

**Comment by s0cks** [comment]
> Thanks for this. And welcome back, hope all has been well.
> 
> Looks like builds failing due to missing license headers, just run `./gradlew updateLicenses` to update them and commit them back up.

Sure thing, thanks!

Was looking for something to code-related to work on in some new found spare time!

I'll take a look at cleaning this up tonight

File: src/main/java/com/atlauncher/gui/components/CollapsiblePanel.java (hunk 2 above)
[the remark this answers is not in the document]
Comment (RyanTheAllmighty) [reply src/main/java/com/atlauncher/gui/components/CollapsiblePanel.java:134]: I don't mind. If it was just something for testing then can just remove it. Play time likely isn't something many people want to see anyway I'd imagine

**Review by RyanTheAllmighty — APPROVED** [review]

**Comment by RyanTheAllmighty** [comment]
Thanks for this, and sorry for the delay in getting this merged. But looks good to me

**Pull request merged**